### PR TITLE
added script for running phylophlan

### PR DESCRIPTION
**Pull request by qiyunzhu**
@ekopylova I added PhyloPhlAn support.

### CODE REVIEW

**Comment by qiyunzhu** [comment]
Also modified the T-REX script.

**Comment by qiyunzhu** [comment]
@sjanssen2 @mortonjt @tkosciol Thanks for your kind offer! Here is the latest PR. It is a Bash script that calls a third-party program (PhyloPhlAn).

File: benchmark/run_phylophlan.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+fi
+
+# set up
+mkdir -p ${working_dir}/phylophlan
+PWD=$(pwd)
+cd ${working_dir}/phylophlan
+mkdir -p input
+mkdir -p output
+mkdir -p temp
+ln -s ${phylophlan_install_dir}/data
```
Comment (tkosciolek) [inline benchmark/run_phylophlan.sh:49]: this line does not have a target

File: benchmark/run_phylophlan.sh [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+TIMEFORMAT='%U %R'
+TIME="$( time ($cmd 1>>$stdout 2>>$stderr) 2>&1)"
+user_time=$(echo $TIME | awk '{print $1}')
+wall_time=$(echo $TIME | awk '{print $2}')
+echo "Total user time PhyloPhlAn: ${user_time}" >> $stderr
+echo "Total wall time PhyloPhlAn: ${wall_time}" >> $stderr
+
+# tear down
+cd $PWD
+
```
Comment (tkosciolek) [inline benchmark/run_phylophlan.sh]: there is no tearing down part. where do you delete temp, etc.?

File: benchmark/run_phylophlan.sh [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+# output genome tree will be:
+# ${working_dir}/phylophlan/output/genomes.tree.nwk
```
Comment (tkosciolek) [inline benchmark/run_phylophlan.sh]: maybe you can print it as string in verbose mode?

File: benchmark/run_trex.sh [hunk 4]
```diff
@@ -24,14 +24,19 @@ args=(
 )
 get_args "$@"
 
+$verbose && echo "Running T-REX .."
+cmd="${trex_install_dir}/hgt3.4 -inputfile=${input_file_nwk}"
+$verbose && echo "Command:"$'\n'"  $cmd"
+
+printf "#TREX\n" >> $output_fp
```
Comment (tkosciolek) [inline benchmark/run_trex.sh:31]: I think you should use `echo -e`

File: benchmark/run_trex.sh [hunk 5]
```diff
@@ -24,14 +24,19 @@ args=(
...
+touch ${stdout%.*}.total_results.txt
+
 TIMEFORMAT='%U %R'
 total_user_time_trex="0.0"
 total_wall_time_trex="0.0"
```
Comment (tkosciolek) [inline benchmark/run_trex.sh:36]: why are you creating those 2 variables if you never use them?

File: benchmark/run_trex.sh [hunk 6]
```diff
@@ -24,14 +24,19 @@ args=(
...
-i=0
 
-printf "#TREX\n" >> $output_fp
-touch ${stdout%.*}.total_results.txt
 # search for HGTs in each gene tree
+i=0
```
Comment (tkosciolek) [inline benchmark/run_trex.sh:39]: you're not incrementing this interator anywhere

File: benchmark/run_trex.sh [hunk 7]
```diff
@@ -24,14 +24,19 @@ args=(
...
 for gene_tree in $gene_tree_dir/*.nwk
 do
     gene_tree_file=$(basename $gene_tree)
```
Comment (tkosciolek) [inline benchmark/run_trex.sh]: you're not using `gene_tree_file` anywhere

**Comment by qiyunzhu** [comment]
@tkosciol Thank you! These are valuable comments!

File: benchmark/run_trex.sh (hunk 7 above)
Comment (qiyunzhu) [reply benchmark/run_trex.sh]: Removed.

File: benchmark/run_trex.sh (hunk 6 above)
Comment (qiyunzhu) [reply benchmark/run_trex.sh:39]: It is in line #59.

File: benchmark/run_trex.sh (hunk 5 above)
Comment (qiyunzhu) [reply benchmark/run_trex.sh:36]: They are used in lines #56-57.

File: benchmark/run_trex.sh (hunk 6 above)
Comment (tkosciolek) [reply benchmark/run_trex.sh:39]: shoot, sorry! GitHub hid those lines from me and I didn't notice. :/

File: benchmark/run_phylophlan.sh (hunk 3 above)
Comment (qiyunzhu) [reply benchmark/run_phylophlan.sh]: This script will be connected to another script, which hasn't been implemented yet. So I left it as is.

File: benchmark/run_phylophlan.sh (hunk 2 above)
Comment (qiyunzhu) [reply benchmark/run_phylophlan.sh]: No they are not deleted... We haven't decided how to deal with them yet. @ekopylova What do you think?

File: benchmark/run_phylophlan.sh (hunk 1 above)
Comment (qiyunzhu) [reply benchmark/run_phylophlan.sh:49]: `ln` will retain the original file name if there is no target.

File: benchmark/run_trex.sh (hunk 4 above)
Comment (qiyunzhu) [reply benchmark/run_trex.sh:31]: This is a good point. Not making any changes but `printf` sends a smell  of C/Perl complication. I replaced all `printf`s with `echo` or `echo -e`. @ekopylova If you have other considerations please let us know.

**Comment by qiyunzhu** [comment]
Also we discussed about two alternative ways of using `$verbose`. Solution A (4 lines):
```
if [ "$verbose" == "true" ]
then
    echo "Command:"$'\n'"  $cmd"
fi
```
Solution 2 (1 line):
```
$verbose && echo "Command:"$'\n'"  $cmd"
```
The latter will require that I modify the `get_args` function in `utils.sh` to force $verbose to be true or false.
Which one do you like? @ekopylova

File: benchmark/run_trex.sh (hunk 4 above)
Comment (qiyunzhu) [reply benchmark/run_trex.sh:31]: @tkosciol I was about to change, but I noticed some issues. See this [post](http://stackoverflow.com/questions/525872/echo-tab-characters-in-bash-script) for people who prefer `printf` over `echo`.
For example, originally it was:
```
printf "$i\t$gene_number\t" >> $output_fp
```
If I change it into `echo`, it becomes:
```
echo $i$'\t'$gene_number$'\t' >> $output_fp
```
Sounds bit awkward. What do you think?

File: benchmark/run_trex.sh (hunk 6 above)
Comment (qiyunzhu) [reply benchmark/run_trex.sh:39]: No problem :)

File: benchmark/run_phylophlan.sh (hunk 2 above)
Comment (mortonjt) [reply benchmark/run_phylophlan.sh]: :+1: for deleting tmp.  You want to avoid bloat, since that filesystems will crash if there are too many files.

Comment (qiyunzhu) [reply benchmark/run_phylophlan.sh]: @mortonjt Thanks! Deleted. By the way, how do you think of the $verbose issue (see end of the page)?

**Comment by qiyunzhu** [comment]
Would anyone like to give some final comments and/or merge? @mortonjt @tkosciol @ekopylova

File: benchmark/run_phylophlan.sh [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+# ----------------------------------------------------------------------------
+
+# usage: run PhyloPhlAn to build phylogenetic tree of whole genomes
+
+# prerequisites:
+# one needs to install the development version of PhyloPhlAn:
+#   hg clone https://bitbucket.org/nsegata/phylophlan
+#   cd phylophlan
+#   hg up dev
+# one needs the follow programs installed and callable from $PATH:
```
Comment (ekopylova) [inline benchmark/run_phylophlan.sh:18]: any specific versions?

**Review by ekopylova — COMMENTED** [review]
@qiyunzhu one comment, otherwise looks good!

File: benchmark/run_phylophlan.sh (hunk 8 above)
Comment (qiyunzhu) [reply benchmark/run_phylophlan.sh:18]: This is the development version Nicola and Francesco are working on. After talking to them they showed me these commands. They said depite not having a version number, this version is pretty stable. They expect to resume the development from April to add some major improvements. @ekopylova

**Comment by ekopylova** [comment]
@qiyunzhu  thanks!

**Comment by qiyunzhu** [comment]
@ekopylova Thank you for merging!

**Pull request merged**